server.js

```diff
@@ -60,16 +89,21 @@ app.all('*', function (req, res, next) {
 
 
 app.get('/authenticate/:code', function(req, res) {
-  console.log('authenticating code:' + req.params.code);
+  log('authenticating code:', req.params.code, true);
   authenticate(req.params.code, function(err, token) {
-    var result = err || !token ? {"error": "bad_code"} : { "token": token };
-    console.log(result);
+    if ( err || !token ) {
+      result = {"error": "bad_code"};
+      log(result.error);
```

Comment:
This is only indirectly related to what your PR fixes, but since we're touching this code, one thing I'd find useful is, in the case where we are returning `err`, logging that instead of just logging `bad_code`. This would make it easier to debug, since it would likely return error messages from whatever Oauth provider this is attached to.

Reply:
I added  change for this.

However, while testing, I noticed your implementation in the authenticate function swallows quite a lot of errors. I opened issue #36 for that. I would prefer that to be fixed in another pr.